Include/internal/pycore_list.h

```diff
@@ -10,6 +10,7 @@ extern "C" {
 
 PyAPI_FUNC(PyObject*) _PyList_Extend(PyListObject *, PyObject *);
 extern void _PyList_DebugMallocStats(FILE *out);
+extern PyObject* _PyList_GetItemRef(PyObject *, Py_ssize_t i);
```

Comment:
Can you add a short comment describing `_PyList_GetItemRef` and how it differs from `PyList_GetItemRef`?
```suggestion
extern PyObject* _PyList_GetItemRef(PyListObject *, Py_ssize_t i);
```

I also think it would make sense for this function to take a `PyListObject*` instead of a `PyObject*`; we're only using it in a context where we've already checked the type. Up to you though.